pkg/utils/asynciter.go

```diff
@@ -156,22 +166,23 @@ func (ic *AsyncItemsCursor) NextItem() (v3io.Item, error) {
 	}
 
 	getItemsResp := resp.Output.(*v3io.GetItemsOutput)
-	shard := resp.Context.(int)
-	resp.Release()
+	defer resp.Release()
```

Comment:
Shouldn't this statement appear earlier in the flow? It seems to me like a leak is possible (e.g. if we get 404 and recurse several lines earlier. Not a regression here, but could be another issue.